src/mame/drivers/albazc.cpp

```diff
@@ -207,12 +207,12 @@ static INPUT_PORTS_START( hanaroku )
 	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_HANAFUDA_NO )
 
 	PORT_START("IN2")   /* 0xe002 */
-	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_SERVICE3 ) PORT_NAME("Data Clear")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_SERVICE4 ) PORT_NAME("Data Clear")
 	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_TILT )
-	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON9 ) PORT_NAME("Medal In") PORT_CODE(KEYCODE_I)
-	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_BUTTON10 ) PORT_NAME("Pay Out") PORT_CODE(KEYCODE_O)
-	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_SERVICE1 ) PORT_NAME("Ext In 1")
-	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_SERVICE2 ) PORT_NAME("Ext In 2")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_MAHJONG_I ) PORT_NAME("Medal In")
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_MAHJONG_O ) PORT_NAME("Pay Out")
```

Comment:
They do not appear to be user controls. 
ie. Going into dip switch service mode - Card 1-6, Play, Start, Yes, No & D-Up - Will Register "PANEL OK" - So thats all the play requires.

I thought "Medal" could be referring to "Tokens", But credits appear to be covered with standard coin 1 & 2 inputs.

I'm not really sure how these ports are supposed to work.
  "Medal In" triggers "DIS" & "CLK" in the diagnostics screen.
  And "Pay Out" causes a "MEDAL EMPTY' message (Once > one credit is inserted).
Neither seem to alter credits (Perhaps it's waiting for a hopper response?).

I'd be inclined to set them as IPT_OTHER until their true nature is known.
Any suggestions on how to investigate this further?